### PR TITLE
kernel: disable CONFIG_CPU_LITTLE_ENDIAN in generic config

### PR DESCRIPTION
**Pull request by abajk**
Endianness depends on CPU architecture. CONFIG_CPU_(BIG/LITTLE)_ENDIAN should
be enabled on target or subtarget based on SoC architecture.

Fixes warning:
```
$ make kernel_oldconfig CONFIG_TARGET=subtarget
...
.config:1008:warning: override: CPU_LITTLE_ENDIAN changes choice state
....
```

Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>

### CODE REVIEW

**Comment by abajk** [comment]
Rebased to current master.

**Comment by adschm** [comment]
I generally agree with your point. However, what happens to the targets that do not set either value if we now explicitly add "CONFIG_CPU_LITTLE_ENDIAN is not set" to generic config (e.g. armvirt, gemini, ...)?

File: target/linux/malta/be/config-default [hunk 1]
```diff
@@ -1,4 +1,3 @@
 CONFIG_CPU_BIG_ENDIAN=y
-# CONFIG_CPU_LITTLE_ENDIAN is not set
```
Comment (adschm) [inline target/linux/malta/be/config-default:2]: For malta, this has been used for 11 (!) years. This does not mean it's a good idea, but it certainly breaks your "with 5.4 it was different" argument in the commit message.

**Comment by Ansuel** [comment]
@abajk still interested?

**Comment by abajk** [comment]
> @abajk still interested?

@Ansuel Yes, I refreshed the commit.

**Comment by Ansuel** [comment]
@abajk thanks for the refresh imho if you want this merged the only way is to add LITTLE-ENDIAN config to every other target that doesn't have BIG-ENDIAN set. Can you do this change? It's the only way to remove it from generic and don't cause any regression.

**Comment by abajk** [comment]
> I generally agree with your point. However, what happens to the targets that do not set either value if we now explicitly add "CONFIG_CPU_LITTLE_ENDIAN is not set" to generic config (e.g. armvirt, gemini, ...)?

The BIG/LITTLe symbols are set only on the MIPS and POWERPC architectures. As far as I can see ARC, ARM and X86 do not require them and they were not set in older versions of OpenWRT (19.07 and 21.02).

**Comment by abajk** [comment]
> @abajk thanks for the refresh imho if you want this merged the only way is to add LITTLE-ENDIAN config to every other target that doesn't have BIG-ENDIAN set. Can you do this change? It's the only way to remove it from generic and don't cause any regression.

Is this necessary? Older versions of OpenWRT did not set the LITTLE_ENDIAN symbol for ARC, ARM and X86 architectures?

**Comment by Ansuel** [comment]
@abajk i'm checking the 2 config CONFIG_CPU_LITTLE_ENDIAN and CONFIG_CPU_BIG_ENDIAN and it's not a generic flag but target specific... 
https://cateee.net/lkddb/web-lkddb/CPU_BIG_ENDIAN.html
https://cateee.net/lkddb/web-lkddb/CPU_LITTLE_ENDIAN.html

We should set it for just the relevant target... for everything else we can skip both of them.

**Comment by abajk** [comment]
@Ansuel 

> We should set it for just the relevant target... for everything else we can skip both of them.

I think I added the symbols to the relevant targets.

Summary:

- ARC - only the CONFIG_CPU_BIG_ENDIAN symbol is defined for this architeture. If it is disabled then the processor operates in LITTLE_ENDIAN mode (default),
- ARM32 - CONFIG_CPU_LITTLE_ENDIAN symbol available since kernel 5.19. This option should be enabled after OpenWRT moves to kernel 6.x. After refreshing the kernel, the symbol disappears,
- ARM64 - enabled CONFIG_CPU_LITTLE_ENDIAN,
- MIPS - enabled relevant symbols,
- POWERPC -  enabled CONFIG_CPU_BIG_ENDIAN,
- UML - Symbols are not defined for this architecture,
- X86 - always little endian. Symbols are not defined for this architecture.

**Comment by Ansuel** [comment]
@abajk thanks! nice work. Need to find some time to check this better but LGTM.

**Comment by Ansuel** [comment]
@abajk i think this is ready... just one last thing... can you put the summary in the commit description?

**Comment by abajk** [comment]
> @abajk i think this is ready... just one last thing... can you put the summary in the commit description?

@Ansuel done

**Pull request merged**